### PR TITLE
Nbo/migrate to m UI 6 prod 13950

### PR DESCRIPTION
**Pull request by nborde-CSM**
Before creating a PR, please check that:
- [ ] Code is clean
- [ ] Tests are still working (cypress tests and `yarn test`)
- [ ] Changes don't cause new errors or warnings in browser console
- [ ] Documentation is up-to-date
- [ ] Breaking changes are clearly identified with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)

### CODE REVIEW

**Review by esasova — COMMENTED** [review]
I have a question about `adaptV4Theme()` function in  _src/theme/index.js_ that is deprecated since a while, maybe it's the right time to get rid of it ?

File: src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericDateInput.js [hunk 1]
```diff
@@ -47,6 +56,8 @@ GenericDateInput.propTypes = {
   setParameterValue: PropTypes.func.isRequired,
   isDirty: PropTypes.bool,
   error: PropTypes.object,
+  gridItemProps: PropTypes.object,
+  size: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (nborde-CSM) [reply src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericDateInput.js:60]: I think this is a script add that I didn't noticed.

File: src/components/ScenarioParameters/ScenarioParameters.js [hunk 2]
```diff
@@ -154,15 +143,19 @@ const ScenarioParameters = ({ onToggleAccordion, isAccordionExpanded }) => {
...
+            <Grid
+              container
+              size="grow"
+              sx={{ direction: 'row', alignItems: 'center', justifyContent: 'space-between' }}
+            >
+              <Grid sx={{ ml: '10px' }}>
                 <Typography>{t('genericcomponent.text.scenario.parameters.title', 'Scenario parameters')}</Typography>
               </Grid>
-              <Grid item>
+              <Grid container>
```
[the remark this answers is not in the document]
Comment (nborde-CSM) [reply src/components/ScenarioParameters/ScenarioParameters.js:158]: ScenarioActions got himself a Grid at his root. It is for this reason that I set container on the last Grid in ScenarioParameters.

File: src/views/Workspaces/Workspaces.js [hunk 3]
```diff
@@ -92,29 +92,33 @@ const Workspaces = () => {
...
+          <Grid container style={{ padding: '18px', height: '90%' }} sx={{ justifyContent: 'center' }}>
+            <Grid size={12}>
               {/* Keep Accordion always open while we have only one organization, and reset default cursor */}
               <Accordion expanded={true} sx={{ '& .MuiAccordionSummary-root:hover': { cursor: 'default !important' } }}>
                 <AccordionSummary>
                   <Typography variant="body1">{organizationName}</Typography>
                 </AccordionSummary>
                 <AccordionDetails>
-                  <Grid container item justifyContent="flex-start" spacing={2} style={{ padding: '24px' }}>
+                  <Grid container spacing={2} style={{ padding: '24px' }} sx={{ justifyContent: 'flex-start' }}>
```
[the remark this answers is not in the document]
Comment (nborde-CSM) [reply src/views/Workspaces/Workspaces.js:121]: I didn't understand why the migration script let it like that...

File: src/views/DatasetManager/components/CreateDatasetButton/components/DatasetWizard/DatasetWizard.js [hunk 4]
```diff
@@ -181,10 +178,8 @@ export const DatasetWizard = ({ open, closeDialog, onConfirm, dataSourceRunTempl
       <Dialog open={open} fullWidth data-cy="dataset-creation-dialog">
         <DialogTitle>{dialogTitle}</DialogTitle>
         <DialogContent>
-          <Grid container gap={1}>
-            <Grid item xs={12}>
-              {stepper}
-            </Grid>
+          <Grid container size="grow" sx={{ gap: 1, width: '100%' }}>
```
[the remark this answers is not in the document]
Comment (nborde-CSM) [reply src/views/DatasetManager/components/CreateDatasetButton/components/DatasetWizard/DatasetWizard.js:181]: I think it is a mess during migration from Grid to Grid 2. I checked and nor` width: 100%` nor `size="grow" ` is usefull on his line. The `size="grow"` in the `Grid` of the next line is sufficient.

**Review by esasova — APPROVED** [review]
LGTM :+1:

File: src/views/DatasetManager/components/CreateDatasetButton/components/DatasetCreationParameters/DatasetCreationParameters.js [hunk 5]
```diff
@@ -84,37 +84,40 @@ export const DatasetCreationParameters = ({ dataSourceRunTemplates, parentDatase
...
-                />
+                <Grid sx={{ pt: 2 }}>
+                  <GenericMultiSelect
+                    parameterData={parameter}
+                    context={{ editMode: true, targetDatasetId: parentDataset?.id }}
+                    parameterValue={value}
+                    setParameterValue={onChange}
+                    isDirty={null}
+                  />
+                </Grid>
```
[the remark this answers is not in the document]
Comment (nborde-CSM) [reply src/views/DatasetManager/components/CreateDatasetButton/components/DatasetCreationParameters/DatasetCreationParameters.js:116]: I don't know if it work. I tried to fix a display glitch "shrink" and I found this solution. I don't remember if I tried the "gridItemProps"...

**Pull request merged**